tensorflow/compiler/tf2tensorrt/convert/ops/slice_ops.cc

```diff
@@ -130,11 +134,31 @@ Status ConvertStridedSliceHelper(
                                   op_instance);
   ITensorProxyPtr tensor = (*slice)->getOutput(0);
 
-  // Reshape for shrink_axis.
+  // Reshape for shrink axis, ellipsis masks based on the shape computed by
+  // ValidateStridedSliceOp or HandleDynamicStridedSliceInput.
+  nvinfer1::Dims dims = tensor->trt_tensor()->getDimensions();
+  std::vector<int> slice_input_dims(dims.d, dims.d + dims.nbDims);
+  StridedSliceShapeSpec empty_spec;
+  empty_spec.shrink_axis_dense_mask = 0;
+  auto shrink_axis_mask = strided_slice_spec.value_or(empty_spec).shrink_axis_dense_mask;
   if (final_shape) {
-    TF_RETURN_IF_ERROR(PrepareTensorForShape(
-        params->converter, TRT_TensorOrWeights(tensor), *final_shape,
-        /*validation_only=*/false, &tensor, node_def, op_instance));
+    int shrink_idx = params->use_implicit_batch ? 1 : 0;
+    if(shrink_axis_mask)  {
+      const auto bShrink_axis_mask = std::bitset<32>(shrink_axis_mask);
+      for(int idx = 0; idx < slice_input_dims.size(); ++idx, ++shrink_idx) {
+        const bool shrink_axis = bShrink_axis_mask[shrink_idx];
+        if(shrink_axis){
+          slice_input_dims[idx] = 0;
+        }
+      }
+      TF_RETURN_IF_ERROR(params->converter->SqueezeTensor(tensor,&slice_input_dims,
+                      params, &tensor, op_instance));
+   }
+   else {
+        TF_RETURN_IF_ERROR(PrepareTensorForShape(
```

Comment:
Is this else branch necessary? I believe `new_axis_mask `is not supported, therefore we only need reshape if we are squeezing away axis. (Is that correct?) And `PrepareTensorForShape `would not work for dynamic input dims.

Reply:
Currently tests related to Unpack and Split fail when I remove the if-else. Is it okay if I take this as a feedback for my upcoming PR?

Reply:
Yes, it  is fine to address this in the next PR.